src/java.base/share/classes/java/time/format/DateTimeFormatterBuilder.java

```diff
@@ -2600,17 +2600,15 @@ private PadPrinterParserDecorator(DateTimePrinterParser printerParser, int padWi
         @Override
         public boolean format(DateTimePrintContext context, StringBuilder buf) {
             int preLen = buf.length();
-            if (printerParser.format(context, buf) == false) {
+            if (!printerParser.format(context, buf)) {
                 return false;
             }
             int len = buf.length() - preLen;
             if (len > padWidth) {
                 throw new DateTimeException(
                     "Cannot print as output of " + len + " characters exceeds pad width of " + padWidth);
             }
-            for (int i = 0; i < padWidth - len; i++) {
-                buf.insert(preLen, padChar);
-            }
+            buf.insert(preLen, String.valueOf(padChar).repeat(padWidth - len));
```

Comment:
Have you checked with a microbenchmark that this added allocation can be elided by JITs and that there's a significant speed-up with your changes? I don't have the necessary domain expertise to assert anything here but I suspect padding widths are typically short. Such as 2 or 4 (for day/month and year fields) so a micro should examine there's no regression for little to no padding. Unlike the original code you call `insert` even if `padWidth - len == 0`. This might be optimized away by JITs, but it'd be good to verify which is best.

Reply:
The modified code is called only when a user explicitly calls `padNext(int, char)`, i.e. if I modified the example snippet as
```java
DateTimeFormatter dtf = new DateTimeFormatterBuilder()
  .appendLiteral("Date:")
//.padNext(20, ' ')
  .append(DateTimeFormatter.ISO_DATE)
  .toFormatter();
String text = dtf.format(LocalDateTime.now());
```
there's no regression.

Right now I cannot build ad-hoc JDK with my changes and check it with JMH, so I'd convert this to draft until I can verify it

Reply:
Meant that you should verify that something like this, which just add a little padding, doesn't regress with your changes: 
```java
DateTimeFormatter dtf = new DateTimeFormatterBuilder()
    .appendLiteral("Year:")
    .padNext(5)
    .appendValue(ChronoField.YEAR)
    .toFormatter();
... 
dtf.format(LocalDateTime.now());
```
And similar for effectively no padding (`.padNext(4)` in the above example). As this API might often be used to ensure short 2-4 char fields are correctly padded I think it's important that we're not adding overhead to such use cases.

Reply:
Added benchmark for this

Reply:
Special casing for len == 0 and keeping the existing buf.insert for len == 1 would avoid object creation except when it would improve performance.

Reply:
@RogerRiggs sorry I don't get it. Maybe you mean speacial casing for `padWidth - len`?

Reply:
Yes, I meant on the length of the inserted padding.

Reply:
Done